### ISSUE TITLE
Partial Reports Uploaded to ReportPortal when Tests are executed in Batches

### ISSUE THREAD

**Issue by sudarsunperu** [Invictum/serenity-reportportal-integration#111]
Hi @Invictum 
Thank you fixing #104 in short notice. I updated to `v.1.5.0` and ran my tests in Batch Execution and now i'm able to see only 1 Launch for all 5 batch executions. The properties that i've passed are as follows
```
gradle :click:test -Dserenity.batch.count=5 -Dserenity.batch.number=1 -Dserenity.rp.communication.dir=sync-dir -Dserenity.rp.modules.count=5 '-Dtags=@TEST_CLICKT-872 or @TEST_CLICKT-8014 or @TEST_CLICKT-878 or @TEST_CLICKT-866 or @TEST_CLICKT-8188' --info
gradle :click:test -Dserenity.batch.count=5 -Dserenity.batch.number=2 -Dserenity.rp.communication.dir=sync-dir -Dserenity.rp.modules.count=5 '-Dtags=@TEST_CLICKT-872 or @TEST_CLICKT-8014 or @TEST_CLICKT-878 or @TEST_CLICKT-866 or @TEST_CLICKT-8188' --info
gradle :click:test -Dserenity.batch.count=5 -Dserenity.batch.number=3 -Dserenity.rp.communication.dir=sync-dir -Dserenity.rp.modules.count=5 '-Dtags=@TEST_CLICKT-872 or @TEST_CLICKT-8014 or @TEST_CLICKT-878 or @TEST_CLICKT-866 or @TEST_CLICKT-8188' --info
gradle :click:test -Dserenity.batch.count=5 -Dserenity.batch.number=4 -Dserenity.rp.communication.dir=sync-dir -Dserenity.rp.modules.count=5 '-Dtags=@TEST_CLICKT-872 or @TEST_CLICKT-8014 or @TEST_CLICKT-878 or @TEST_CLICKT-866 or @TEST_CLICKT-8188' --info
gradle :click:test -Dserenity.batch.count=5 -Dserenity.batch.number=5 -Dserenity.rp.communication.dir=sync-dir -Dserenity.rp.modules.count=5 '-Dtags=@TEST_CLICKT-872 or @TEST_CLICKT-8014 or @TEST_CLICKT-878 or @TEST_CLICKT-866 or @TEST_CLICKT-8188' --info
```
All 5 tests are executed by the 5 batches (1 each batch) and the Serenity Report confirms the same. **However now I find only 3 (instead of 5) test case results on Report Portal.** The `sync-dir` contains 1 file with 0 bytes after the end of the Pipeline Execution.
![image](https://user-images.githubusercontent.com/18679720/85109658-f4d4db00-b211-11ea-8b0d-2b2de787393b.png)
![image](https://user-images.githubusercontent.com/18679720/85109827-3e252a80-b212-11ea-9f2f-2c84917ae004.png)
**ReportPortal Launches tab**
![image](https://user-images.githubusercontent.com/18679720/85110238-da4f3180-b212-11ea-8a70-3b4ed3881881.png)

Can you please help with this?

**Comment by Invictum** [comment]
Hi @sudarsunperu 

One file in directory indicates that merge mechanism wasn't executed properly. When all works as expected sync directory should be removed automatically. Gradle commands looks good though.

1. How do you run that batches in parallel or one by one?
2. Make sure sync directory is persistent and accessible for all runs.
3. It would be nice to have a simple project that reproduces an issue.

**Comment by sudarsunperu** [comment]
Hi @Invictum 

> 
> 1. How do you run that batches in parallel or one by one?
**I run the batches in parallel.** 
> 2. Make sure sync directory is persistent and accessible for all runs.
**All 5 batches are started on the same Jenkins Executor and in the same stage at the same time (more or less). The `sync-dir` hence in the same workspace and for sure is accessible by all the batches.**
> 3. It would be nice to have a simple project that reproduces an issue.
**I will try to work on that and update it here. However I hope the answers to the other questions help with this issue.**

**Comment by Invictum** [comment]
@sudarsunperu 
Unfortunately I'm unable to reproduce issue you mentioned. So at this stage synthetic projects will be rally appreciated.

**Comment by grey-rain** [comment]
abandoned.
